pkg/chunk/gcp/storage_client.go

```diff
@@ -34,57 +35,82 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.instance, "bigtable.instance", "", "Bigtable instance ID.")
 }
 
-// storageClient implements chunk.storageClient for GCP.
-type storageClient struct {
+type storageClientV1 struct {
+	storageClientV2
+}
+
+// storageClientV2 implements chunk.storageClientV2 for GCP.
+type storageClientV2 struct {
 	cfg       Config
 	schemaCfg chunk.SchemaConfig
 	client    *bigtable.Client
 }
 
 // NewStorageClient returns a new StorageClient.
 func NewStorageClient(ctx context.Context, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.StorageClient, error) {
+	return NewStorageClientV2(ctx, cfg, schemaCfg)
+}
+
+// NewStorageClientV1 returns a new v1 StorageClient.
+func NewStorageClientV1(ctx context.Context, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.StorageClient, error) {
+	client, err := bigtable.NewClient(ctx, cfg.project, cfg.instance, instrumentation()...)
+	if err != nil {
+		return nil, err
+	}
+	return &storageClientV1{
+		storageClientV2{
+			cfg:       cfg,
+			schemaCfg: schemaCfg,
+			client:    client,
+		},
+	}, nil
+}
+
+// NewStorageClientV2 returns a new v2 StorageClient.
+func NewStorageClientV2(ctx context.Context, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.StorageClient, error) {
 	client, err := bigtable.NewClient(ctx, cfg.project, cfg.instance, instrumentation()...)
 	if err != nil {
 		return nil, err
 	}
-	return &storageClient{
+	return &storageClientV2{
 		cfg:       cfg,
 		schemaCfg: schemaCfg,
 		client:    client,
 	}, nil
 }
 
-func (s *storageClient) NewWriteBatch() chunk.WriteBatch {
-	return bigtableWriteBatch{
+func (s *storageClientV2) NewWriteBatch() chunk.WriteBatch {
+	return bigtableWriteBatchV2{
 		tables: map[string]map[string]*bigtable.Mutation{},
 	}
 }
 
-type bigtableWriteBatch struct {
+type bigtableWriteBatchV2 struct {
 	tables map[string]map[string]*bigtable.Mutation
 }
 
-func (b bigtableWriteBatch) Add(tableName, hashValue string, rangeValue []byte, value []byte) {
+func (b bigtableWriteBatchV2) Add(tableName, hashValue string, rangeValue []byte, value []byte) {
 	rows, ok := b.tables[tableName]
 	if !ok {
 		rows = map[string]*bigtable.Mutation{}
 		b.tables[tableName] = rows
 	}
 
-	// TODO the hashValue should actually be hashed - but I have data written in
-	// this format, so we need to do a proper migration.
-	rowKey := hashValue + separator + string(rangeValue)
+	hasher := sha256.New()
+	hasher.Write([]byte(hashValue))
```

Comment:
SHA256 is pretty slow, and we don't need its cryptographic properties.  I'd have probably picked FNV (fast and pretty uniform), and then append the row key to final hashed value to guarantee uniqueness.